dev-tools/mage/dashboard.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mage
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/magefile/mage/sh"
+)
+
+// ExportDashboard exports a dashboard from Kibana and writes it into the given module.
+func ExportDashboard() error {
+	module := EnvOr("MODULE", "")
+	if module == "" {
+		return fmt.Errorf("MODULE must be specified")
+	}
+
+	id := EnvOr("ID", "")
+	if id == "" {
+		return fmt.Errorf("Dashboad ID must be specified")
+	}
+
+	beatsDir, err := ElasticBeatsDir()
+	if err != nil {
+		return err
+	}
+
+	// TODO: This is currently hardcoded for KB 6, we need to figure out what we do for KB 7
+	file := filepath.Join(beatsDir, BeatName, "module", module, "_meta/kibana/6/dashboard", id+".json")
```

Comment:
How does this work with dashboards at locations like https://github.com/elastic/beats/tree/master/heartbeat/monitors/active/http/_meta/kibana/6/dashboard ?

Should those move to this new simpler pattern?

Reply:
It doesn't at the moment. We should probably make the path a config param which each beat can set, so it also work here (follow up PR).

Reply:
For the output file you don’t want to use a path that is relative to the `ElasticBeatsDir` because this won’t work for community beats or x-pack beats. Instead use `CWD()` or just leave it as a relative path assuming the tool this is being passed to does the write thing and writes the output based on the CWD that it was executed from.

So in short, use `file := CWD(“module”, module, “_meta/kibana/6/dashboard”, id+”.json”)`.